src/program/command_mode.rs

```diff
@@ -98,26 +97,32 @@ fn glob_path(path: &str) -> Result<Vec<PathBuf>, String> {
                 push_image_path(&mut new_images, p);
             }
             Err(e) => {
-                let err_msg = format!("Error: Unexpected path {}", e);
+                let err_msg = format!("Unexpected path {}", e);
                 return Err(err_msg);
             }
         }
     }
     if new_images.is_empty() {
-        let err_msg = format!("Error: path \"{}\" had no images", path);
+        let err_msg = format!("Path \"{}\" had no images", path);
         return Err(err_msg);
     }
     Ok(new_images)
 }
 
-/// Uses regex to parse user input, then concating all subsequent strings that are escaped together
-/// TODO: have regex capture the ending space in case of escaped files
+/// Uses regex to parse user input that's provided to command mode into arguments
+/// First argument is the command followed by its arguments
 fn parse_user_input(input: &str) -> Vec<String> {
     let mut input_vec: Vec<String> = Vec::new();
-    let re: Regex = Regex::new(r"[\x21-\x7E]+(\\)?").unwrap();
+    // Regex pattern matches all unicode characters there must be at least 1
+    // and the '\' character may be present at the end of the string
+    // This is used to rather than split on whitespace, only get the actual arguments
+    let regex = match Regex::new(r"[\x21-\x7E]+(\\)?") {
```

Comment:
Yep, with the cross platform approach I'm thinking about this would be a global constant in `command_mode.rs`. 

Coming up with really good names is always hard...

Reply:
So, how would I type this as as newglob?
`/home/alex/test dir/**/*.png`

would it be
`ng /home/alex/test\ dir/**/*.png`?

Reply:
Yeah, because otherwise it's impossible to know the end of a file name without making syscalls for every string provided and their combinations.

Funny enough, this is a bug in Neovim, and I'm not making an issue because it's impossible to fix...

make a file `~/home/$USER/test\\`
open neovim
:e test\\ Documents/*
current buffer: `test\\ Documents/*`

This results in whenever you try to save the current buffer it fails the write operation because that's never allow to be a file which is just awesome. 

In our case we don't have that bug because of cli.rs/push_image_path just gives back nothing...

Reply:
![gotten_glob](https://user-images.githubusercontent.com/28968644/58051234-41ce1580-7b17-11e9-87dd-62b75215deeb.png)

Reply:
Does `ng` with a directory with spaces work for you?

Reply:
It does not, actually need to remove the `\`s in order for it to be recognized as a dir. Thanks for the catch, I completely neglected that.

Reply:
path_regex would be a fine name, and of course, it should only be compiled once, even though the overhead is presumably tiny.

Reply:
A regex was an over complicated approach that was naive and shortsighted, for instance our input to command mode may not always a be a file. Each command should handle and parse its own input since it can be interpreted differently in this context. 

Marking as resolved.